test/Microsoft.Azure.Devices.Edge.Test.Common/linux/PackageManagement.cs

```diff
@@ -0,0 +1,187 @@
+                            "set -e",
+                            $"{this.forceInstallConfigCmd} {string.Join(' ', packages)}",
+                            $"{this.packageTool} {this.installCmd}"
+                        };
+                case PackageManagement.SupportedPackageExtension.Rpm:
+                    return new[]
+                        {
+                            "set -e",
+                            $"{this.packageTool} {this.installCmd} {string.Join(' ', packages)}",
+                            "pathToSystemdConfig=$(systemctl cat iotedge | head -n 1); sed 's/=on-failure/=no/g' ${pathToSystemdConfig#?} > ~/override.conf; sudo mv -f ~/override.conf ${pathToSystemdConfig#?}; sudo systemctl daemon-reload;"
+                        };
+                default:
+                    throw new NotImplementedException($"Don't know how to install daemon on for '.{this.packageExtension}'");
+            }
+        }
+
+        public async Task<string[]> GetInstallCommandsFromMicrosoftProdAsync()
+        {
+            await this.CheckInitializationAsync();
+
+            switch (this.packageExtension)
+            {
+                case PackageManagement.SupportedPackageExtension.Deb:
+                    // Based on instructions at:
+                    // https://github.com/MicrosoftDocs/azure-docs/blob/058084949656b7df518b64bfc5728402c730536a/articles/iot-edge/how-to-install-iot-edge-linux.md
+                    // TODO: 8/30/2019 support curl behind a proxy
+                    return new[]
+                        {
+                            $"curl https://packages.microsoft.com/config/{this.os}/{this.version}/multiarch/prod.list > /etc/apt/sources.list.d/microsoft-prod.list",
+                            "curl https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor > /etc/apt/trusted.gpg.d/microsoft.gpg",
+                            $"{this.packageTool} update",
+                            $"{this.packageTool} install --yes iotedge"
+                        };
+                case PackageManagement.SupportedPackageExtension.Rpm:
+                    return new[]
+                        {
+                            $"{this.forceInstallConfigCmd} https://packages.microsoft.com/config/{this.os}/{this.version}/packages-microsoft-prod.rpm",
+                            $"{this.packageTool} updateinfo",
+                            $"{this.packageTool} install --yes iotedge",
+                            "pathToSystemdConfig=$(systemctl cat iotedge | head -n 1); sed 's/=on-failure/=no/g' ${pathToSystemdConfig#?} > ~/override.conf; sudo mv -f ~/override.conf ${pathToSystemdConfig#?}; sudo systemctl daemon-reload;"
+                        };
+                default:
+                    throw new NotImplementedException($"Don't know how to install daemon on for '.{this.packageExtension}'");
+            }
+        }
+
+        public async Task InternalStopAsync(CancellationToken token)
```

Comment:
This should be called `StopAsync` since it's a public method.

Reply:
done